### PR TITLE
Update API documentation

### LINKED ISSUE #941: Document missing JSON-RPC APIs
For example `miner_setCoinbase` is currently not documented.

We should review the list of methods [here](https://github.com/hyperledger/besu/tree/main/ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/methods) and determine what else is missing.

### PR DESCRIPTION
**Pull request by rolandtyler**
Signed-off-by: Roland Tyler <roland.tyler@consensys.net>

## Describe the change
Added missing JSON-API methods to docs:
* `debug_accountAt`
*  `miner_setCoinbase` /  `miner_setEtherbase`
* `trace_call `
*  `trace_filter `

## Issue fixed
Fixes #941 

### For content changes

- [x] Doc content
- [ ] Doc pages organisation

### For tools changes

- [ ] CircleCI workflow
- [ ] Build and QA tools (lint, vale,…)
- [ ] MkDocs templates
- [ ] MkDocs configuration
- [ ] Python dependencies
- [ ] Node dependencies and JavaScript
- [ ] ReadTheDocs configuration
- [ ] GitHub integration

### CODE REVIEW

File: docs/Reference/API-Methods.md [hunk 1]
```diff
@@ -787,6 +787,59 @@ We recommend using the [`TRACE` API](#trace-methods) for production use over the
     methods, use the [`--rpc-http-api`](CLI/CLI-Syntax.md#rpc-http-api) or
     [`--rpc-ws-api`](CLI/CLI-Syntax.md#rpc-ws-api) options.
 
+### `debug_accountAt`
+
+Returns an account information at the given specified index in the specified block.
```
Comment (macfarla) [inline docs/Reference/API-Methods.md:792]: s/an account/account

File: docs/Reference/API-Methods.md [hunk 2]
```diff
@@ -787,6 +787,59 @@ We recommend using the [`TRACE` API](#trace-methods) for production use over the
...
+
+#### Parameters
+
+* `blockHashOrNumber`: *string* - block hash or number
+
+* `txIndex`: *number* - transaction index from which to start
+
+* `address`: *string* - address hash from which to start
```
Comment (macfarla) [inline docs/Reference/API-Methods.md:800]: account address for which to retrieve information

File: docs/Reference/API-Methods.md [hunk 3]
```diff
@@ -7052,6 +7146,152 @@ the returned list items include the [revert reason](../HowTo/Send-Transactions/R
...
+
+`result`: *array* of *objects* - list of [calls to other contracts](Trace-Types.md#trace) containing
+one object per call, in transaction execution order
+
+!!! example
+
+    === "curl HTTP request"
+
+        ```bash
+        curl -X POST --data '{"jsonrpc":"2.0","method":"trace_call","params":[{"fromBlock":"0x1","toBlock":"0x21","after":2,"count":2,"fromAddress":["0xfe3b557e8fb62b89f4916b721be55ceb828dbd73"]}],"id":415}' http://127.0.0.1:8545
```
Comment (macfarla) [inline docs/Reference/API-Methods.md:7233]: this is still trace_call

File: docs/Reference/API-Methods.md [hunk 4]
```diff
@@ -787,6 +787,59 @@ We recommend using the [`TRACE` API](#trace-methods) for production use over the
...
+
+Returns account information at the given specified index in the specified block.
+
+#### Parameters
+
+* `blockHashOrNumber`: *string* - block hash or number
+
+* `txIndex`: *number* - transaction index from which to start
+
+* `address`: *string* - account address for which to retrieve information
```
[the remark this answers is not in the document]
Comment (rolandtyler) [reply docs/Reference/API-Methods.md]: Confirmed, it should be contract address.

File: docs/Reference/API-Methods.md [hunk 5]
```diff
@@ -787,6 +787,59 @@ We recommend using the [`TRACE` API](#trace-methods) for production use over the
...
+    === "curl HTTP request"
+
+        ```bash
+        curl -X POST --data '{"jsonrpc":"2.0","method":"debug_accountAt","params":["0xc8df1f061abb4d0c107b2b1a794ade8780b3120e681f723fe55a7be586d95ba6", 0, "0xbcde5374fce5edbc8e2a8697c15331677e6ebf0b"],"id":1}' http://127.0.0.1:8545
+        ```
+
+    === "wscat WS request"
+
+        ```bash
+        {"jsonrpc":"2.0","method":"debug_accountAt","params":[""0xc8df1f061abb4d0c107b2b1a794ade8780b3120e681f723fe55a7be586d95ba6", 0, "0xbcde5374fce5edbc8e2a8697c15331677e6ebf0b"],"id":1}
```
Comment (macfarla) [inline docs/Reference/API-Methods.md:825]: there is an extra " before the blockhash

File: docs/Reference/API-Methods.md [hunk 6]
```diff
@@ -787,6 +787,59 @@ We recommend using the [`TRACE` API](#trace-methods) for production use over the
...
+
+        ```json
+        {
+          "jsonrpc": "2.0",
+          "id": 1,
+          "result": {
+            "code": "0x0",
+            "nonce": "0x0",
+            "balance": "0x340aad21b3b70000",
+            "codehash" : "0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470"
```
Comment (macfarla) [inline docs/Reference/API-Methods.md:838]: I think that you would either get code (and codehash ie if it's a contract address you get both code related fields together), or balance (if it's an externally owned account) - but I don't think it makes sense to have codehash and balance. Have you verified this result?

Comment (macfarla) [reply docs/Reference/API-Methods.md:838]: I stand corrected. 
This is what I get for an externally owned account
```
{
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
        "code": "0x0",
        "nonce": "0x5",
        "balance": "0xad78ebc5ac6200000",
        "codehash": "0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470"
    }
}
```

Comment (macfarla) [reply docs/Reference/API-Methods.md:838]: and actually I get an internal error if I supply a contract address. I think this is a bug... will follow up!

Comment (macfarla) [reply docs/Reference/API-Methods.md:838]: ok I have filed a bugfix here https://github.com/hyperledger/besu/pull/3518 and this is what I get for the result if it's a contract address. I think it's useful to show the two examples. 
```
{
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
        "code": "0x608060405234801561001057600080fd5b506004361061002b5760003560e01c8063b27b880414610030575b600080fd5b61004a60048036038101906100459190610108565b61004c565b005b60606000806000604051935036600085376000803686885af490503d9150816000853e806000811461007d57610093565b60008311156100925761012085019350836040525b5b5060008114156100ec578473ffffffffffffffffffffffffffffffffffffffff167f410d96db3f80b0f89b36888c4d8a94004268f8d42309ac39b7bcba706293e099856040516100e3919061016e565b60405180910390a25b5050505050565b60008135905061010281610227565b92915050565b60006020828403121561011e5761011d610211565b5b600061012c848285016100f3565b91505092915050565b600061014082610190565b61014a818561019b565b935061015a8185602086016101de565b61016381610216565b840191505092915050565b600060208201905081810360008301526101888184610135565b905092915050565b600081519050919050565b600082825260208201905092915050565b60006101b7826101be565b9050919050565b600073ffffffffffffffffffffffffffffffffffffffff82169050919050565b60005b838110156101fc5780820151818401526020810190506101e1565b8381111561020b576000848401525b50505050565b600080fd5b6000601f19601f8301169050919050565b610230816101ac565b811461023b57600080fd5b5056fea2646970667358221220fdfb5c371055342507b8fb9ca7b0c234f79819bd5cb05c0d467fb605de979eb564736f6c63430008060033",
        "nonce": "0x1",
        "balance": "0x0",
        "codehash": "0xf5f334d41776ed2828fc910d488a05c57fe7c2352aab2d16e30539d7726e1562"
    }
}
```

File: docs/Reference/API-Methods.md [hunk 7]
```diff
@@ -787,6 +787,90 @@ We recommend using the [`TRACE` API](#trace-methods) for production use over the
...
 
+### `debug_accountAt`
+
+Returns account information at the given specified index of the specified block.
+
+#### Parameters
+
+* `blockHashOrNumber`: *string* - block hash or number
+
+* `txIndex`: *number* - transaction index from which to start
```
Comment (macfarla) [inline docs/Reference/API-Methods.md:798]: transaction index at which to retrieve account information

File: docs/Reference/API-Methods.md [hunk 8]
```diff
@@ -787,6 +787,90 @@ We recommend using the [`TRACE` API](#trace-methods) for production use over the
     methods, use the [`--rpc-http-api`](CLI/CLI-Syntax.md#rpc-http-api) or
     [`--rpc-ws-api`](CLI/CLI-Syntax.md#rpc-ws-api) options.
 
+### `debug_accountAt`
+
+Returns account information at the given specified index of the specified block.
+
+#### Parameters
+
+* `blockHashOrNumber`: *string* - block hash or number
```
Comment (macfarla) [inline docs/Reference/API-Methods.md:796]: block hash or number at which to retrieve account information

**Review by macfarla — APPROVED** [review]
couple of final minor clarifications, otherwise LGTM!

File: docs/Reference/API-Methods.md [hunk 9]
```diff
@@ -787,6 +787,90 @@ We recommend using the [`TRACE` API](#trace-methods) for production use over the
     methods, use the [`--rpc-http-api`](CLI/CLI-Syntax.md#rpc-http-api) or
     [`--rpc-ws-api`](CLI/CLI-Syntax.md#rpc-ws-api) options.
 
+### `debug_accountAt`
+
+Returns account information at the given specified index of the specified block.
```
Comment (alexandratran) [inline docs/Reference/API-Methods.md:792]:
```suggestion
Returns account information at the specified index of the specified block.
```

File: docs/Reference/API-Methods.md [hunk 10]
```diff
@@ -796,9 +880,9 @@ Returns the accounts for a specified block.
 
 #### Parameters
 
-* `blockHashOrNumber`: *string* - block hash or number
+* `blockHashOrNumber`: *string* - block hash or number at which to retrieve account information
 
-* `txIndex`: *number* - transaction index from which to start
+* `txIndex`: *number* - transaction index at which to retrieve account information
 
 * `address`: *string* - address hash from which to start
```
Comment (alexandratran) [inline docs/Reference/API-Methods.md:887]: Should these match the parameter descriptions for `txIndex` and `address` for `debug_accountAt`?

File: docs/Reference/API-Methods.md [hunk 11]
```diff
@@ -5105,10 +5189,51 @@ command line option.
...
+Sets the coinbase, the address for the mining rewards.
+
+!!! note
+
+    You can also use `miner_setEtherbase` as an alternative method. They both work the same way.
+    Etherbase is a historic name for coinbase.
+
+#### Parameters
+
+`coinbase`: *string* - The account address you pay mining rewards to
```
Comment (alexandratran) [inline docs/Reference/API-Methods.md:5203]:
```suggestion
`coinbase`: *string* - account address you pay mining rewards to
```

for consistency with other param descriptions on this page

File: docs/Reference/API-Methods.md [hunk 12]
```diff
@@ -5105,10 +5189,51 @@ command line option.
...
+
+#### Returns
+
+`result`: *boolean* - `true` when address is set.
```
Comment (alexandratran) [inline docs/Reference/API-Methods.md:5207]:
```suggestion
`result`: *boolean* - `true` when address is set
```

for consistency

**Review by alexandratran — APPROVED** [review]
Looks good! Some minor suggestions.

**Pull request merged**